### PR TITLE
DoT Support

### PR DESCRIPTION
**Pull request by nils-wisiol**
This adds DoT support, but without authentication. This is along the lines of [draft-ietf-dprive-unilateral-probing](https://www.ietf.org/archive/id/draft-ietf-dprive-unilateral-probing-01.html#name-server-name-indication). While not published yet, it is already adopted by the IETF.

For DoQ, see https://github.com/PowerDNS/pdns/issues/9897 and [OARC 38](https://indico.dns-oarc.net/event/43/contributions/918/).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -38,8 +38,8 @@ services:
     depends_on:
     - ns
     ports:
-    - "53:53"
-    - "53:53/udp"
+    - "${DESEC_NS_PUBLIC_PORT-53}:53"
+    - "${DESEC_NS_PUBLIC_PORT-53}:53/udp"
```
Comment (peterthomassen) [inline docker-compose.yml]: `:-53`?

File: dnsdist/conf/dnsdist.conf [hunk 2]
```diff
@@ -23,8 +23,8 @@ addAction("internal.", RCodeAction(DNSRCode.REFUSED))
 --  for why a password is used instead of an API key.
 webserver("10.16.4.10:8083")
 setWebserverConfig({
-    password="we+ensure+security+via+network+segmentation",
-    apiKey="",
+    password="$scrypt$ln=10,p=1,r=8$pKoROW2iHTUujxtGGX7y/Q==$h3+OfsjwnslAcGMcHkO48j+5Ss7nry+WraiXEGIB1SM=",
+    apiKey="$scrypt$ln=10,p=1,r=8$pKoROW2iHTUujxtGGX7y/Q==$h3+OfsjwnslAcGMcHkO48j+5Ss7nry+WraiXEGIB1SM=",
```
Comment (peterthomassen) [inline dnsdist/conf/dnsdist.conf:27]: needs to be what's in https://github.com/desec-io/desec-ns/blob/99f6cadf4f1a9824420ca9f792b0770ab8f91c21/prometheus/conf/prometheus.yml#L23

**Review by peterthomassen — CHANGES_REQUESTED** [review]
awesome!

**Comment by nils-wisiol** [comment]
Try DoT query locally with `kdig SOA @127.0.0.1 +tls _signal.$DESEC_NS_NAME` (`DESEC_NS_NAME` from .env)

**Comment by peterthomassen** [comment]
also, 079ac776be0be31ce8f21d91a6a4e732a287d7c6 fixes #41

File: docker-compose.yml (hunk 1 above)
Comment (nils-wisiol) [reply docker-compose.yml]: done

File: dnsdist/conf/dnsdist.conf (hunk 2 above)
Comment (nils-wisiol) [reply dnsdist/conf/dnsdist.conf:27]: this is the hash of a random password. I reverted to `hashPassword()` for `password=` and kept the hash for `apiKey`.

**Review by peterthomassen — APPROVED** [review]
lgtm!

**Pull request merged**